plugins/Live/Model.php

```diff
@@ -104,7 +104,7 @@ public function queryLogVisits($idSite, $period, $date, $segment, $offset, $limi
             }
         }
 
-        if ($checkforMoreEntries) {
+        if ((int)$limit > -1 && $checkforMoreEntries) {
```

Comment:
I guess this shouldn't be done here. Let's assume we set `$checkforMoreEntries=true`, then we expect to receive an array like `[array, boolean]`. If `$limit=-1`, we now would return only the result array, which would be kind of unexpected behavior. Simply reverting the change here should do the trick. The number of found visits can't ever be `-1`, so it will in this case always return `[array, false]`, which is correct, as there can't be more visits when looking without limit.

Reply:
I see, thanks. Now I know why the tests are failing. I will look at it tomorrow.